builds/e2e/longhaul.yaml

```diff
@@ -346,4 +346,128 @@ jobs:
           snitch.storage.account: '$(snitch.storage.account)'
           snitch.storage.masterKey: '$(StorageAccountMasterKeyStress)'
           loadGen.message.frequency: '$(loadGen.message.frequency.amd64)'
-         
\ No newline at end of file
+          
+################################################################################
+  - job: setup_iotuap_amd64_moby
+################################################################################
+    displayName: Windows IoT Core AMD64 Moby (Setup Only)
+    condition: eq(variables['run.iotuap.amd64.moby'], 'true')
+    pool:
+      name: $(pool.name)
+      demands:
+        - longhaul_iotuap_x64 -equals true
```

Comment:
please refer to this OneNote page, https://microsoft.sharepoint.com/teams/Azure_IoT/_layouts/OneNote.aspx?id=%2Fteams%2FAzure_IoT%2FShared%20Documents%2FEngineering%2FIoTHubNotes&wd=target%28Azure%20IoT%20Edge%2FWiki.one%7C70253E27-7EC5-43AD-89E4-C9BA3943D8CF%2FIoT%20Edge%20VMs%5C%2FDevices%7C0330C378-E885-4582-A9ED-C80CBEA3F05F%2F%29
onenote:https://microsoft.sharepoint.com/teams/Azure_IoT/Shared%20Documents/Engineering/IoTHubNotes/Azure%20IoT%20Edge/Wiki.one#IoT%20Edge%20VMs/Devices&section-id={70253E27-7EC5-43AD-89E4-C9BA3943D8CF}&page-id={0330C378-E885-4582-A9ED-C80CBEA3F05F}&end and add/update for newly created machines.

Currently we use "run-long-haul" capability to indicate that machine is used for long haul with other parameters e.g. 
agent-os-name to distinguish for different platforms.

Reply:
What is the point of using two parameters:
 * run-long-haul=**true**
 * agent-os-name=**WinIoTCore_x64**

instead of just one (that carries information about both the test type and the OS type)?
 * longhaul_iotuap_x64=**true**

Reply:
It is easier to understand and keep consistent with existing tests.

Reply:
Keep in mind that Windows IoT Core is different - unlike the other test setups it needs two VMs (a VSTS agent VM and a runner VM). The runner VM needs to be Windows IoT Core, but the VSTS agent machine can be anything (Windows Pro/Server or Linux), it cannot be Windows IoT Core.
Therefore the condition _agent-os-name=**WinIoTCore_x64**_ won't make much sense.
The only thing we need to guarantee is that we have 1:1 mapping between the Longhaul test pipeline and its dedicated VSTS agent VM that targets the Windows IoT Core runner.

Let me know if you still would like to see this change.

Reply:
Make sense that we should not use agent-os-name; instead we should name it as runner-os-name or target-os-name.

Reply:
Changed to use capabilities _runner-os-name_, _run-stress_ and _run-long-haul_.